exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java

```diff
@@ -138,7 +141,9 @@ public Viewable getPlugins() {
   @Produces(MediaType.APPLICATION_JSON)
   public PluginConfigWrapper getPluginConfig(@PathParam("name") String name) {
     try {
-      return new PluginConfigWrapper(name, storage.getStoredConfig(name));
+      return authEnabled.separateWorkspace()
```

Comment:
On this, and similar lines, it would be better to factor out the logic into a method on `WebUserConnection` such as `getStorage()` or some such. Too much copy/paste and room for error otherwise.